sql: TestCancelDistSQLQuery failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/7ce0cba090528e7c50bd8fe708d536362f5d766a

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stressrace instead of stress and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
make stress TESTS=TestCancelDistSQLQuery PKG=github.com/cockroachdb/cockroach/pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-stderr=false -maxtime 20m -timeout 10m'
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=977304&tab=buildLog

```
I181020 05:44:48.386396 94410 server/server.go:1525  [n2] starting https server at 127.0.0.1:33211 (use: 127.0.0.1:33211)
I181020 05:44:48.386410 94410 server/server.go:1527  [n2] starting grpc/postgres server at 127.0.0.1:36283
I181020 05:44:48.386420 94410 server/server.go:1528  [n2] advertising CockroachDB node at 127.0.0.1:36283
I181020 05:44:48.389210 94410 server/server.go:1581  [n2] done ensuring all necessary migrations have run
I181020 05:44:48.389253 94410 server/server.go:1584  [n2] serving sql connections
I181020 05:44:48.395899 94666 sql/event_log.go:126  [n1,client=127.0.0.1:52506,user=root] Event: "create_database", target: 52, info: {DatabaseName:test Statement:CREATE DATABASE IF NOT EXISTS test User:root}
I181020 05:44:48.399466 94666 sql/event_log.go:126  [n1,client=127.0.0.1:52506,user=root] Event: "create_table", target: 53, info: {TableName:test.public.nums Statement:CREATE TABLE test.public.nums (num INT) User:root}
I181020 05:44:48.407427 94666 storage/replica_command.go:298  [n1,s1,r1/1:/M{in-ax}] initiating a split of this range at key /Table/53/1/50 [r2]
I181020 05:44:48.430162 93201 util/stop/stopper.go:537  quiescing; tasks left:
3      node.Node: batch
1      [async] closedts-rangefeed-subscriber
I181020 05:44:48.430218 93201 util/stop/stopper.go:537  quiescing; tasks left:
1      node.Node: batch
1      [async] closedts-rangefeed-subscriber
W181020 05:44:48.430432 94734 server/node.go:942  [n2] node=2: unable to log node_join event: log-event: failed to send RPC: sending to all 1 replicas failed; last error: <nil> rpc error: code = Unavailable desc = transport is closing
W181020 05:44:48.430530 94874 internal/client/txn.go:532  [n2,hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (error=rpc error: code = Unavailable desc = transport is closing [exhausted])
I181020 05:44:48.430557 94874 storage/node_liveness.go:817  [n2,hb] retrying liveness update after storage.errRetryLiveness: result is ambiguous (error=rpc error: code = Unavailable desc = transport is closing [exhausted])
W181020 05:44:48.430656 94874 internal/client/txn.go:532  [n2,hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W181020 05:44:48.430675 94874 storage/node_liveness.go:494  [n2,hb] failed node liveness heartbeat: node unavailable; try another peer
W181020 05:44:48.430740 94868 ts/db.go:195  [n2,ts-poll] error writing time series data: failed to send RPC: sending to all 1 replicas failed; last error: <nil> rpc error: code = Unavailable desc = transport is closing
I181020 05:44:48.430780 94930 util/stop/stopper.go:537  quiescing; tasks left:
1      ts.poller: poll
1      node.Node: writing summary
1      [async] closedts-rangefeed-subscriber
1      [async] auto-upgrade
W181020 05:44:48.430813 94869 server/node.go:845  [n2,summaries] error recording initial status summaries: failed to send RPC: sending to all 1 replicas failed; last error: <nil> rpc error: code = Unavailable desc = transport is closing
I181020 05:44:48.431048 94930 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber
1      [async] auto-upgrade
I181020 05:44:48.431405 93201 util/stop/stopper.go:537  quiescing; tasks left:
1      node.Node: batch
I181020 05:44:48.432732 94930 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] auto-upgrade
W181020 05:44:48.433289 94898 internal/client/txn.go:532  [n2,intExec=read-setting] failure aborting transaction: node unavailable; try another peer; abort caused by: context canceled
W181020 05:44:48.433440 94737 internal/client/txn.go:532  [n2,intExec=show-version] failure aborting transaction: node unavailable; try another peer; abort caused by: read-setting: context canceled
I181020 05:44:48.433513 94732 server/server_update.go:64  [n2] failed attempt to upgrade cluster version, error: show-version: context canceled
--- FAIL: TestCancelDistSQLQuery (0.33s)
	run_control_test.go:245: pq: cannot up-repliate to s2; missing gossiped StoreDescriptor

```

--- a/pkg/sql/run_control_test.go
+++ b/pkg/sql/run_control_test.go
@@ -236,14 +236,16 @@ func TestCancelDistSQLQuery(t *testing.T) {
 	if _, err := conn1.Exec("ALTER TABLE nums SPLIT AT VALUES (50)"); err != nil {
 		t.Fatal(err)
 	}
-	// Make the second node the leaseholder for the first range to distribute
-	// the query.
-	if _, err := conn1.Exec(fmt.Sprintf(
-		"ALTER TABLE nums EXPERIMENTAL_RELOCATE VALUES (ARRAY[%d], 1)",
-		tc.Server(1).GetFirstStoreID(),
-	)); err != nil {
-		t.Fatal(err)
-	}
+
+	// Make the second node the leaseholder for the first range to distribute the
+	// query. This may have to retry if the second store's descriptor has not yet
+	// propagated to the first store's StorePool.
+	testutils.SucceedsSoon(t, func() error {
+		_, err := conn1.Exec(fmt.Sprintf(
+			"ALTER TABLE nums EXPERIMENTAL_RELOCATE VALUES (ARRAY[%d], 1)",
+			tc.Server(1).GetFirstStoreID()))
+		return err
+	})
 
 	// Run queryToCancel to be able to get an estimate of how long it should
 	// take. The goroutine in charge of cancellation will sleep a random

--- a/pkg/storage/replica_command.go
+++ b/pkg/storage/replica_command.go
@@ -1137,7 +1137,7 @@ func (s *Store) AdminRelocateRange(
 			for _, candidate := range candidates {
 				store, ok := storeMap[candidate.StoreID]
 				if !ok {
-					return fmt.Errorf("cannot up-repliate to s%d; missing gossiped StoreDescriptor", candidate.StoreID)
+					return fmt.Errorf("cannot up-replicate to s%d; missing gossiped StoreDescriptor", candidate.StoreID)
 				}
 				candidateDescs = append(candidateDescs, *store)
 			}

